lvsfunc/dehalo.py

```diff
@@ -203,31 +212,33 @@ def fine_dehalo(clip: vs.VideoNode, ref: vs.VideoNode | None = None,
-                            Must be between 0 and 100.
-    :param highsens:        Upper sensitivity range. The higher this is, the more it will process.
-                            Must be between 0 and 100.
-    :param rfactor:         Image enlargement factor. Set to >1 to enable some form of aliasing-protection.
-                            Must be greater than 1.
-    :param show_mask:       Return mask clip. Valid options are 1–7.
-    :param planes:          Specifies which planes will be processed.
-                            Any unprocessed planes will be simply copied.
-
-    :return:                Dehalo'd clip or halo mask clip.
+    :param clip:                    Clip to process.
+    :param ref:                     Reference clip. Will replace regular dehaloing.
+    :param rx:                      Horizontal radius for halo removal. Must be greater than 1. Will be rounded up.
+    :param ry:                      Vertical radius for halo removal. Must be greater than 1. Will be rounded up.
+    :param brightstr:               Strength for bright halo removal.
+    :param darkstr:                 Strength for dark halo removal. Must be between 0 and 1.
+    :param thmi:                    Minimum threshold for sharp edges. Keep only the sharpest edges (line edges).
+                                    To see the effects of this setting take a look at the strong mask (show_mask=4).
+    :param thma:                    Maximum threshold for sharp edges. Keep only the sharpest edges (line edges).
+                                    To see the effects of this setting take a look at the strong mask (show_mask=4).
+    :param thlimi:                  Minimum limiting threshold. Includes more edges than previously,
+                                    but ignores simple details.
+    :param thlima:                  Maximum limiting threshold. Includes more edges than previously,
+                                    but ignores simple details.
+    :param lowsens:                 Lower sensitivity range. The lower this is, the more it will process.
+                                    Must be between 0 and 100.
+    :param highsens:                Upper sensitivity range. The higher this is, the more it will process.
+                                    Must be between 0 and 100.
+    :param rfactor:                 Image enlargement factor. Set to >1 to enable some form of aliasing-protection.
+                                    Must be greater than 1.
+    :param show_mask:               Return mask clip. Valid options are 1–7.
+    :param planes:                  Specifies which planes will be processed.
+                                    Any unprocessed planes will be simply copied.
+
+    :return:                        Dehalo'd clip or halo mask clip.
+
+    :raises ModuleNotFoundError:    Dependencies are missing.
     """
     warnings.warn("fine_dehalo: 'This function is deprecated in favor of `vsdehalo.fine_dehalo`! "
```

Comment:
You put a warning here but in bidehalo you only put a warning in the docstrings, what's the way to deprecate a function then?

Reply:
They both have the same warnings in the docstring and when the user calls the function